libraries/rush-lib/src/logic/ProjectWatcher.ts

```diff
@@ -139,13 +156,17 @@ export class ProjectWatcher {
 
         this._hasRenderedStatus = false;
 
-        const resolveIfChanged = async (): Promise<void> => {
+        this._resolveIfChanged = async (): Promise<void> => {
```

Comment:
```suggestion
        const resolveIfChanged = this._resolveIfChanged = async (): Promise<void> => {
```
Then most of the plumbing complexity goes away since it is still bound in scope.

Also at the end of the `await new Promise` do `.finally(() => this._resolveIfChanged = undefined)`